### PR TITLE
SM-353 Implement Sylo Staking Manager

### PR DESCRIPTION
**Pull request by JCSanPedro**
Implements the interface and contract for Sylo Staking.

### CODE REVIEW

File: test/staking/syloStakingManager.test.ts [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+      amount,
+      0n,
+    );
+  };
+
+  /**
+   * testStakingAction is a utility function that asserts a user and node's
+   * stake entry is correctly updated after performing a staking related
+   * transaction. The user's balance is also validate that it has been updated
+   * correctly as well.
```
Comment (Nick95550) [inline test/staking/syloStakingManager.test.ts:525]:
```suggestion
   * transaction. The user's balance is also validated that it has been updated
   * correctly as well.
```

**Review by Nick95550 — CHANGES_REQUESTED** [review]
A couple comments

File: test/staking/syloStakingManager.test.ts [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+
+    assert.equal(
+      unlocking.amount,
+      500n,
+      'Expected each unlocking to add to total unlocking amount',
+    );
+  });
+
+  it('sets unlock at based on unlock duration', async () => {
+    const { node, user } = await setupInitialStake(1000n);
```
[the remark this answers is not in the document]
Comment (JCSanPedro) [reply test/staking/syloStakingManager.test.ts:183]: The name of the field is actually `unlockAt`

File: test/staking/syloStakingManager.test.ts [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+    const nodeStake = await syloStakingManager.getManagedStake(
+      node,
+      user.getAddress(),
+    );
+
+    assert.equal(nodeStake.amount, 334n, "Expected node's stake to decrease");
+  });
+
+  it('can reverts when transferring stake to nil address', async () => {
+    const { node, user } = await setupInitialStake(1000n);
```
Comment (Nick95550) [inline test/staking/syloStakingManager.test.ts:469]: nit: 
```suggestion
  it('reverts when transferring stake to nil address', async () => {
    const { node, user } = await setupInitialStake(1000n);
```

**Review by Nick95550 — APPROVED** [review]
one little thing otherwise LGTM

**Pull request merged**